packages/plugins/openapi/src/rest-generator.ts

```diff
@@ -907,7 +907,7 @@ export class RESTfulOpenAPIGenerator extends OpenAPIGeneratorBase {
         if (mode === 'create') {
             // 'id' is required if there's no default value
             const idFields = model.fields.filter((f) => isIdField(f));
-            if (idFields.length && idFields.every((f) => !hasAttribute(f, '@default'))) {
+            if (idFields.length === 1 && idFields.every((f) => !hasAttribute(f, '@default'))) {
```

Comment:
shall we simplify this to `hasAttribute(idFields[0], '@default')`?

Reply:
That's not the same behaviour though? That would set the to level required 'id' for compound ids as well.

Reply:
Oh, sorry I meant:

```ts
idFields.length === 1 && !hasAttribute(idFields[0], '@default')
```

Just feeling the `.every` call is not needed anymore